### ISSUE TITLE
Give editor resizables a max/min size

### ISSUE THREAD

**Issue by DanielBreiner** [motion-canvas/motion-canvas#492]
**Describe the bug**
The editor allows the user to resize panels without limit. Which also allows them to destroy the layout completely, creating overflowing and overlapping elements.
The user might want to hide a panel, but they should never stretch it so it messes up a different panel.

**To Reproduce**
Drag the separator between panels to stretch one so much, it messes up with the layout of the second one.

**Expected behavior**
The inspector (left panel) should have a minimum width. When user drags the separator too far left, it should snap to a 0 width.
The preview window should have a minimum width. I don't see a point in ever hiding it completely.
The timeline should have a maximum height (or rather the rest of the editor a minimum height). I don't think the timeline needs a minimum height and it definitely makes sense to hide it completely (e.g. for presentation mode).

Hopefully this covers all cases and makes sense. If not, please provide feedback and discuss in the comments.

**Package versions:**

- ui: 3.1.0

**Screenshots**

![image](https://user-images.githubusercontent.com/43221175/223576783-23816b44-0c1e-4a16-a146-2cd44010f45a.png)
![image](https://user-images.githubusercontent.com/43221175/223576849-d02bd02e-32e5-4df3-8f7a-b82f687427d4.png)
![image](https://user-images.githubusercontent.com/43221175/223576718-83e018b3-dd97-47c1-b2b2-81ed39e87989.png)

**Comment by aarthificial** [comment]
I don't really understand why this is a bug. 
The interface does not get "destroyed", it continues to function correctly. There are no overlapping or overflowing elements, everything stays contained within the panels. Certain elements get clipped because of the lack of space but that's not what *overflowing* or *overlapping* means. 
It's also possible to resize the layout back without any problem.

Users can easily hide the side panel by unselecting the current tab:
![image](https://user-images.githubusercontent.com/64662184/223582349-d31e1cdb-bf01-4957-9aba-0099360ee521.png)

Here's WebStorm, allowing me to resize the project panel however I like:
![image](https://user-images.githubusercontent.com/64662184/223582708-f6a91607-a89b-4c4e-a440-575908d28dd2.png)

Here's Blender also letting me do this with the panel on the right:
![image](https://user-images.githubusercontent.com/64662184/223582879-b75e948b-d3fa-48d2-afda-fdbafed37350.png)

TLDR: I don't see the point in imposing some artificial restrictions on users without a good reason.

**Comment by DanielBreiner** [comment]
For full context -- here's what I was proposing originally:

https://user-images.githubusercontent.com/43221175/223746247-ba749b8f-3480-4189-a361-7fd8cd9e6328.mov

**Comment by aarthificial** [comment]
Seeing this in action I agree this would be a nice feature to have.
I still wouldn't restrict the size of panels though.
